tests/integration_tests/instances.py

```diff
@@ -1,41 +1,34 @@
 # This file is part of cloud-init. See LICENSE file for license information.
-from abc import ABC, abstractmethod
+from abc import ABC
 import logging
 import os
 from tempfile import NamedTemporaryFile
 
-from pycloudlib import EC2, GCE, Azure, OCI, LXD
-from pycloudlib.cloud import BaseCloud
 from pycloudlib.instance import BaseInstance
 
 import cloudinit
 from cloudinit.subp import subp
 from tests.integration_tests import integration_settings
 
 try:
-    from typing import Callable, Optional
+    from typing import TYPE_CHECKING
+    if TYPE_CHECKING:
+        from tests.integration_tests.clouds import IntegrationCloud
 except ImportError:
     pass
 
 
 log = logging.getLogger('integration_testing')
 
 
-class IntegrationClient(ABC):
-    client = None  # type: Optional[BaseCloud]
-    instance = None  # type: Optional[BaseInstance]
-    datasource = None  # type: Optional[str]
+class IntegrationInstance(ABC):
```

Comment:
Without any abstract members, this is no longer an ABC:

```suggestion
class IntegrationInstance:
```